### PR TITLE
Add changes to boots...

### PR DESCRIPTION
**Pull request by ptrivedi**
... for
* custom registry and registry certificate specification
* OSIE path override
* eliminating registry user/pass requirement

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/tinkerbell/boots/patches/0001-Add-environment-variables-for.patch [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+ func mustBuildOSIEURL() *url.URL {
+ 	u, err := buildOSIEURL()
+ 	if err != nil {
+@@ -55,8 +75,10 @@ func buildWorkerParams() {
+ 	dockerRegistry = getParam("DOCKER_REGISTRY")
+ 	grpcAuthority = getParam("TINKERBELL_GRPC_AUTHORITY")
+ 	grpcCertURL = getParam("TINKERBELL_CERT_URL")
+-	registryUsername = getParam("REGISTRY_USERNAME")
+-	registryPassword = getParam("REGISTRY_PASSWORD")
++	registryUsername = os.Getenv("REGISTRY_USERNAME")
```
Comment (pokearu) [inline projects/tinkerbell/boots/patches/0001-Add-environment-variables-for.patch:119]: The `getParam()` method does call `os.Getenv()`
Was there a reason to not use it?
https://github.com/tinkerbell/boots/blob/v0.6.0/installers/osie/mirror.go#L63

File: projects/tinkerbell/boots/patches/0001-Add-environment-variables-for.patch [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
++	registryPassword = os.Getenv("REGISTRY_PASSWORD")
++	registryCertUrl = os.Getenv("REGISTRY_CERT_URL")
++	tinkWorkerImage = os.Getenv("TINK_WORKER_IMAGE")
+ }
+ 
+ func getParam(key string) string {
+diff --git a/packet/mock_cacher/cacher_mock.go b/packet/mock_cacher/cacher_mock.go
+index a12fadc..88fe3c7 100644
```
Comment (pokearu) [inline projects/tinkerbell/boots/patches/0001-Add-environment-variables-for.patch:127]: Look like all the changes in this file are punctuation. In that case can we not patch this file? Just to track the changes better.
Also maybe running `make golangci-lint` on boots might remove the changes?

File: projects/tinkerbell/boots/patches/0001-Add-environment-variables-for.patch (hunk 1 above)
Comment (ptrivedi) [reply projects/tinkerbell/boots/patches/0001-Add-environment-variables-for.patch:119]: One of the purposes of this PR is to make registry username and password optional. getParam has a check to make sure they are not blank.

File: projects/tinkerbell/boots/patches/0001-Add-environment-variables-for.patch (hunk 2 above)
Comment (ptrivedi) [reply projects/tinkerbell/boots/patches/0001-Add-environment-variables-for.patch:127]: Hmm yeah can revert. This is mocks auto-generated stuff

**Review by pokearu — APPROVED** [review]

**Review by abhinavmpandey08 — APPROVED** [review]
/lgtm

**Pull request merged**